src/shared/components/palette/collection/write-collection-field.component.ts

```diff
@@ -48,15 +48,19 @@ export class WriteCollectionFieldComponent extends AbstractFieldWriteComponent i
   }
 
   buildCaseField(item, index: number): CaseField {
-    return {
+    return this.newCaseField(index, item);
+  }
+
+  private newCaseField(index: number, item) {
+    return Object.assign(new CaseField(), {
```

Comment:
just came to my mind rather than Object.assign why not simply: 
```cf = new CaseField();
cf.id = index.ToString();
cf.field_type =  this.caseField.field_type.collection_field_type;
...

```

Reply:
Depends on how strongly you feel about it,

Reply:
ok let's leave it